### PR TITLE
Add utility functions for comparison and rounding

### LINKED ISSUE #1622: Add utility functions for comparison and rounding
It would be nice to have a couple extra utility functions for RationalTime:
* A function that compares whether two RationalTimes have the same value and rate. This is different from the "==" operator which scales the time before comparison (e.g., ```otime::RationalTime(30.2) == otime::RationalTime(60.4, 2.0)```).
* Rounding functions that return the ```floor()```, ```ceil()```, and ```round()``` of a RationalTime.

### PR DESCRIPTION
**Pull request by darbyjohnston**
Fixes #1622

This PR adds a few utility functions to RationalTime for comparison and rounding:
* A function that compares whether two RationalTimes have the same value and rate. This is different from the "==" operator which scales the time before comparison (e.g., otime::RationalTime(30.2) == otime::RationalTime(60.4, 2.0)).
* Rounding functions that return the floor(), ceil(), and round() of a RationalTime.

If these seem useful I can also add Python bindings for them.

### CODE REVIEW

File: src/opentime/rationalTime.h
Comment (meshula) [inline src/opentime/rationalTime.h:90]: I almost wonder, to make the method names more self-describing, if they might be named `floor_to_rate`, `ceil_to_rate`, and `round_to_rate`.

File: src/opentime/rationalTime.h
Comment (meshula) [inline src/opentime/rationalTime.h:83]: I think `exactly_equal` harkens to javascript's `===`. 

If I see a function called `exactly_equal` my mind thinks that this is a strict equality  check, that is not rate conformed, which is correct, of course. I would prefer `strictly_equal` though since `exactly_equal` doesn't tell me whether or not the values are conformed before the comparison, whereas strict equality does. I believe `strict equality` is the modern term of art.

eg. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality

Comment (darbyjohnston) [reply src/opentime/rationalTime.h:83]: Good suggestion, I'll update the name.

File: src/opentime/rationalTime.h
Comment (darbyjohnston) [reply src/opentime/rationalTime.h:90]: The ```_to_rate``` sounds a bit confusing to me, maybe ```floor_value```, ```ceil_value```, and ```round_value```?

Comment (jminor) [reply src/opentime/rationalTime.h:90]: I think `floor_value` is better, since it tells me which field is changing.

Comment (meshula) [reply src/opentime/rationalTime.h:90]: I can go for that :)

**Review by meshula — APPROVED** [review]

**Comment by meshula** [comment]
Looking good to me, maybe get a second approval :)

File: src/opentime/rationalTime.h
Comment (apetrynet) [inline src/opentime/rationalTime.h:270]: Please excuse my lack of C++ foo. 
Is a method prefixed with an "_" meant to be "private" or not encouraged for use outside the class in C++?
I'm asking as replacing this could potentially cause some issues for people using it?

Comment (meshula) [reply src/opentime/rationalTime.h:270]: The underbar means "private" although it's not a general C++ convention. It's common in film, on linux/mac. @darbyjohnston I didn't think about the name when I approved, just the removal of the underbar. Is there a reason to add _value() to all the functions? floor() ceil() and round() are class methods so they won't conflict with the standard library.

Comment (meshula) [reply src/opentime/rationalTime.h:270]: Ah, I see the conversation. For the record, @jminor suggested them, and I said that sounded fine to me :) Josh's reasoning was that floor is taking the floor of the value. But now that time has passed, and I read it without context, value is such a generic term, that it is no longer intuitive to me that floor_value() means take the floor of the value member. I'll defer to Josh on this one ~ I'm sure I will be confused a couple more times until it sticks in my brain.

Comment (darbyjohnston) [reply src/opentime/rationalTime.h:270]: I think that's a point in favor of the shorter names, when you looked at it fresh that's what you expected to see. I personally think the "_value" is implicit and if we need additional rounding functions they can have more explicit names:
```
RationalTime floor() const;
RationalTime floor_rate() const;
RationalTime floor_value_and_rate() const;
```

Comment (darbyjohnston) [reply src/opentime/rationalTime.h:270]: I reverted back to the shorter names, hopefully it reads better and I think operating on the value is what people would expect.

**Comment by meshula** [comment]
Thanks Darby!

**Pull request merged**
